test/classes/initializers/phase1/dynamicThis/castThis.chpl

```diff
@@ -0,0 +1,26 @@
+
+class Parent {
+  var x : int;
+
+  proc init() {
+    this.x = 5;
+  }
+
+  proc inheritedMethod() {
+    writeln("Parent.inheritedMethod()");
+  }
+}
+
+class Child : Parent {
+  var y : real;
+
+  proc init() {
+    this.y = x ** 2;
+    inheritedMethod();
+    (this:Child).inheritedMethod();
```

Comment:
I agree that this cast shouldn't be legal in phase 1.  The presence of the test (and the fact that it fired even when we don't create a `new Child()` suggesting it's a syntactic pattern rather than a semantic one) made me wonder whether the user ought to be able to cast to an ancestor class like `Grandparent` in phase 1 (and/or whether doing so would have any actual utility... I have to admit that I can't quite remember the behavior of a dynamic cast on a class variable in Chapel when the result is non-`nil`).